File: src/main/java/ro/nicolaemariusghergu/queryit/service/MiniMarketService.java
package ro.nicolaemariusghergu.queryit.service;

import org.springframework.http.ResponseEntity;
import ro.nicolaemariusghergu.queryit.dto.MiniMarketDto;

import java.util.List;

public interface MiniMarketService {

    ResponseEntity<MiniMarketDto> findMiniMarketById(Long id);

    ResponseEntity<List<MiniMarketDto>> getMiniMarkets();

    ResponseEntity<Long> addMiniMarket(MiniMarketDto miniMarketDto);

    ResponseEntity<MiniMarketDto> getMiniMarketByName(String name);

    ResponseEntity<Long> deleteMiniMarketById(Long id);

    ResponseEntity<MiniMarketDto> updateMiniMarket(MiniMarketDto miniMarketDto);
}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/service/MiniMarketService.kt
package ro.nicolaemariusghergu.queryit.service

import org.springframework.http.ResponseEntity
import ro.nicolaemariusghergu.queryit.dto.MiniMarketDto

interface MiniMarketService {
    open fun findMiniMarketById(id: Long?): ResponseEntity<MiniMarketDto?>?
    open fun getMiniMarkets(): ResponseEntity<MutableList<MiniMarketDto?>?>?
    open fun addMiniMarket(miniMarketDto: MiniMarketDto?): ResponseEntity<Long?>?
    open fun getMiniMarketByName(name: String?): ResponseEntity<MiniMarketDto?>?
    open fun deleteMiniMarketById(id: Long?): ResponseEntity<Long?>?
    open fun updateMiniMarket(miniMarketDto: MiniMarketDto?): ResponseEntity<MiniMarketDto?>?
}